peter_rockwood/HW1.md

```diff
@@ -0,0 +1,234 @@
+
+### Homework #1
+(Due Monday 1/12)
+
+You need to turn in this homework by cloning this repo, making a new branch with your solutions, pushing to github, and issuing a pull request.
+For details, follow the instructions [here](http://portlandcodeschool.github.io/jse/2015/01/07/command-line-and-git-slides/#/14).
+
+Please also read the [learning objectives](objectives.md) for this week.
+
+---
+
+**1)** (_Difficulty: mixed_)
+Predict the output of each of the following expressions, and explain the rules which determine it.
+If the output depends on the value of x, identify the conditions when the expression will output true.  Assume the cases are independent, and x is reset to an unknown value before each.
+
+Some of these are tricky!  Don't trust your first instinct.  
+
+
+**a)** `"1" == 1`
+
+	True, type-converting equality.     					
+
+**b)** `"1" === 1`
+
+	False, strict equality.    					
+
+**c)** `x == 'x'`  
+
+	True if the value of x is 'x', otherwise false.   					
+
+**d)** `x == (x+'')`  
+
+	True, the empty string is ignored.					
+
+**e)** `'' == ' '`	
+
+	False, a space counts as a character in a string.					
+
+**f)** `x = true`
+
+	True, this is just the value 'true' assigned to x, which returns 'true'.						
+
+**g)** `var x; x == 'undefined'`
+
+	False, an initialized, but unassigned variable has the value and returns 'undefined',
+	but does not have the value of the string 'undefined'		
+
+**h)** `'9'<'10'`	
+
+	False, no type-conversion for inequalities.					
+
+**i)** `typeof x + 1 === "number"`	
+
+	False, typeof x gets evaluated first returning a string then adding 1.	
+
+**j)** `typeof x % 2 === "number"`	
+
+	False, typeof x gets evaluated first returning a string and then applying modulus.	
+
+**k)** `typeof (x % 2) === "number"`	
+
+	True if x happens to be assigned a number.
+
+**l)** `x++ == ++x`	
+
+	False, x++ returns x previous to the add, ++x returns x after the add.
+
+**m)** `++x == x++`	
+
+	True, ++x returns x+1 and assigns it to x. x++ returns the 							same, new value of x. 
+
+**n)** `"1"+x == 1+x`	
+
+	False, "1"+x prepends '1' onto number value x, which will increase the order of x by 1 and never be equal to x+1.				
+
+**o)** `"0"+1 == 1`							
+
+	True, prepending a zero onto 1 still evaluates to 1.
+
+**p)** `(typeof (x+1))===(typeof x)`		
+
+	True, if x is a number. Also true if x is a string, (x+1) will convert x to a string before the expression on the right is evaluated.
+
+**q)** `(x*1 == x) || ((typeof x) != "number")`	
+
+	False, if x is a number, the first condition is true, but the 
+	second condition is false. If x is not a number the same holds 
+	true vice-versa.
```

Comment:
The `||` here is a logical OR rather than a logical AND, which is `&&`. This means that the expression is `true` in virtually all cases, the exception being (again) `NaN`.